internal/engine/execution/procedure.go

```diff
@@ -445,73 +396,24 @@ type evaluatable func(ctx context.Context, exec dbQueryFn, values map[string]any
 // See their execution in (*callMethod).execute inside the `range e.Args` to
 // collect the `inputs` passed to the call of a dataset method or other
 // "namespace" method, such as an extension method.
-func makeExecutables(exprs []tree.Expression, schema *types.Schema) ([]evaluatable, error) {
-	execs := make([]evaluatable, 0, len(exprs))
-
-	for _, expr := range exprs {
-		switch e := expr.(type) {
-		case *tree.ExpressionBindParameter:
-			// This could be a special one that returns an evaluatable that
-			// ignores the passed ResultSetFunc since the value is
-		case *tree.ExpressionTextLiteral, *tree.ExpressionNumericLiteral, *tree.ExpressionBooleanLiteral,
-			*tree.ExpressionNullLiteral, *tree.ExpressionBlobLiteral, *tree.ExpressionUnary,
-			*tree.ExpressionBinaryComparison, *tree.ExpressionFunction, *tree.ExpressionArithmetic:
-			// Acceptable expression type.
-		default:
-			return nil, fmt.Errorf("unsupported expression type: %T", e)
-		}
-
-		// clean expression, since it is submitted by the user
-		err := expr.Walk(clean.NewStatementCleaner(schema.Procedures))
-		if err != nil {
-			return nil, err
-		}
-
-		// The schema walker is not necessary for inline expressions, since
-		// we do not support table references in inline expressions.
-		walker := sqlanalyzer.NewWalkerRecoverer(expr)
-		paramVisitor := parameters.NewParametersWalker()
-		err = walker.Walk(paramVisitor)
-		if err != nil {
-			return nil, fmt.Errorf("error replacing parameters: %w", err)
-		}
-
-		// SELECT expr;  -- prepare new value in receivers for subsequent
-		// statements This query needs to be run in "simple" execution mode
-		// rather than "extended" execution mode, which asks the database for
-		// OID (placeholder types) that it can't know since there's no FOR table.
-		selectTree := &tree.SelectStmt{
-			Stmt: &tree.SelectCore{
-				SimpleSelects: []*tree.SimpleSelect{
-					{
-						SelectType: tree.SelectTypeAll,
-						Columns: []tree.ResultColumn{
-							&tree.ResultColumnExpression{
-								Expression: expr,
-							},
-						},
-					},
-				},
-			},
-		}
-
-		stmt, err := tree.SafeToSQL(selectTree)
-		if err != nil {
-			return nil, err
+func makeExecutables(params []*actions.InlineExpression) ([]evaluatable, error) {
+	var evaluatables []evaluatable
+
+	for _, param := range params {
+		// copy the param to avoid loop variable capture
+		param2 := &actions.InlineExpression{
+			Statement:     param.Statement,
+			OrderedParams: append([]string{}, param.OrderedParams...),
```

Comment:
Is this append for a one line slice copy?  Maybe `slices.Clone` would be more straightforwared, but I think a deep copy of the fields is more than what's needed to avoid the loop variable capture.

Reply:
Did not know about `slices.Clone`.

However you are right, I do not think we actually have to copy the `OrderedParams` slice.